harness/determined/core/_train.py

```diff
@@ -89,10 +95,30 @@ def report_training_metrics(
             data=det.util.json_encode(body),
         )
 
-        # Also sync tensorboard.
-        if self._tbd_writer and self._tbd_mgr:
-            self._tbd_writer.on_train_step_end(steps_completed, metrics, batch_metrics)
-            self._tbd_mgr.sync()
+        if self._tensorboard_mode == TensorboardMode.AUTO:
+            if self._tbd_writer:
+                self._tbd_writer.on_train_step_end(steps_completed, metrics, batch_metrics)
+            self._tensorboard_manager.sync()
+
+    def get_tensorboard_path(self) -> pathlib.Path:
+        """
+        Get TensorBoard log directory path.
+        """
+        return self._tensorboard_manager.base_path
+
+    def upload_tensorboard_files(
+        self,
+        selector: Callable[[pathlib.Path], bool] = lambda _: True,
+        mangler: Callable[[pathlib.Path, int], pathlib.Path] = lambda p, __: p,
+    ) -> None:
+        """
+        Upload files generated for consumption by Tensorboard to checkpoint storage.
+        :param selector: optional function returning True for a file that should be included.
+        If not provided, all files are uploaded.
+        :param mangler: optional function modifying the destination file names based on rank.
+        """
+        logger.info("upload tb profile")
+        self._tensorboard_manager.sync(selector, mangler, self._distributed.rank)
```

Comment:
question: should upload_tensorboard_metrics be allowed in non-AUTO mode?

Reply:
`upload_tensorboard_files` is there specifically for the non-AUTO (MANUAL) mode. I am confused by the question.

Reply:
sorry, I meant to say "in AUTO mode".